### PR TITLE
Codeql analysis

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Code scanning was enabled on the `old-master` branch, but didn't get transferred to `main`. This pull request is a rebase of @clanmills's [original commit](https://github.com/Exiv2/exiv2/commit/83865876ff7370e2aebfb97a07dda332aab91eea) onto `main`. I also added `main` to the list of branches to test.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```
Comment (hassec) [inline .github/workflows/codeql-analysis.yml:41]: There is a warning that this is apparently not needed anymore. 

But @kevinbackhouse you are the CodeQL expert so leaving the best config for you to decide ;)

Comment (kevinbackhouse) [reply .github/workflows/codeql-analysis.yml]: @hassec: Thanks! I have updated it.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [0.27-maintenance, main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [0.27-maintenance, main]
+  schedule:
+    - cron: '0 8 * * 4'
```
Comment (hassec) [inline .github/workflows/codeql-analysis.yml:15]: @kevinbackhouse sorry I missed this part on my first review, but why would we need this scheduled run on top of push & pull request?

Comment (kevinbackhouse) [reply .github/workflows/codeql-analysis.yml:15]: @hassec: Good point. That was cut 'n paste from me, and you're right that we don't need it.

**Review by hassec — APPROVED** [review]
Thanks @kevinbackhouse :+1:

**Pull request merged**